src/api.test.ts

```diff
@@ -1,22 +1,149 @@
+    - height=-500000000
+    - width=hihi
+
+    - 
+    404(Not Found)
+    - /cat
+    - /cat/hello/hi/green/
+    
+4. Authentication
+    403(Unauthenticated)
+ 5. Performance
+
+
+ */
 
 describe("GET /", () => {
-    it("must response with hello world", async () => {
-        const response = await request(app).get('/');
-        expect(response.status).toBe(200);
-        expect(response.body.data).toEqual("HELLO WORLD");
-    })
-    // must response with random image - fixed image
-    // must response with resized image when width and height is specified
-    // must response with image which is corresponding with the random seed
-})
-
-// Grayscale
-// Blur
-// 
-
-// hanyi.com/?width=300&height=200&seed=abcd
-// https://fastly.picsum.photos/id/127/200/300.jpg?hmac=H0aErkmw8FxF1Tp7uFj4cV-aVMxDDjOVKTwGwS6REXw
-// https://picsum.photos/id/237/200/300
-// https://picsum.photos/id/238/200/300
\ No newline at end of file
+  it("must response with a random image", async () => {
+    // mocking response from unsplash api
+    nock("http://localhost:1080")
+      .get("/photos/random?client_id=ACCESS_KEY_SAMPLE")
+      .reply(200, randomSuccessResponse_01);
+
+    // request and getting response from the app
+    const response = await request(app).get("/");
+    expect(response.status).toBe(302);
+
+    // Testing on redirect url
+    expect(response.header.location).toEqual(
+      "https://images.unsplash.com/photo-1692278265511-7c884556cd74?crop=entropy&cs=srgb&fm=jpg&ixid=M3w0OTU5Mzh8MHwxfHJhbmRvbXx8fHx8fHx8fDE2OTUxMTYxNzh8&ixlib=rb-4.0.3&q=85"
+    );
+  });
+
+  it("must response with a random image with size of 400X600", async () => {
+    // mocking response from unsplash api
+    nock("http://localhost:1080")
+      .get("/photos/random?client_id=ACCESS_KEY_SAMPLE")
+      .reply(200, randomSuccessResponse_01);
+
+    // request and getting response from the app
+    const response = await request(app).get("/?width=400&height=600");
+    expect(response.status).toBe(302);
+
+    // Testing on redirect url
+    expect(response.header.location).toEqual(
+      "https://images.unsplash.com/photo-1692278265511-7c884556cd74?crop=entropy&cs=srgb&fm=jpg&ixid=M3w0OTU5Mzh8MHwxfHJhbmRvbXx8fHx8fHx8fDE2OTUxMTYxNzh8&ixlib=rb-4.0.3&q=85&w=400&h=600"
+    );
+  });
+
+  it("must response 400(Bad Request) when width is not a number", async () => {
+    // mocking response from unsplash api
+    nock("http://localhost:1080")
+      .get("/photos/random?client_id=ACCESS_KEY_SAMPLE")
+      .reply(200, randomSuccessResponse_01);
+
+    // request and getting response from the app
+    const response = await request(app).get("/?width=hihi&height=600");
+    expect(response.status).toBe(400);
+
+    // Testing on redirect url
+    expect(response.text).toEqual(
+      '"The value of width and height should be a number"'
+    );
+  });
+
+  it("must response 400(Bad Request) when width is not in the range of number(w > 2000)", async () => {
```

Comment:
seems like it should be
```suggestion
  it("must response 400(Bad Request) when width is not in the range of number(0 < w <= 2000)", async () => {
```